packages/studio-base/src/components/PanelList/index.tsx

```diff
@@ -347,11 +351,21 @@ function PanelList(props: Props): JSX.Element {
       if (mode === "grid") {
         return;
       }
-      if (e.key === "ArrowDown" && highlightedPanelIdx != undefined) {
-        setHighlightedPanelIdx((highlightedPanelIdx + 1) % allFilteredPanels.length);
-      } else if (e.key === "ArrowUp" && highlightedPanelIdx != undefined) {
-        const newIdx = (highlightedPanelIdx - 1) % (allFilteredPanels.length - 1);
-        setHighlightedPanelIdx(newIdx >= 0 ? newIdx : allFilteredPanels.length + newIdx);
+      if (e.key === "ArrowDown") {
+        setHighlightedPanelIdx((existing) => {
+          if (existing == undefined) {
+            return 0;
+          }
+          return (existing + 1) % allFilteredPanels.length;
+        });
+      } else if (e.key === "ArrowUp") {
+        setHighlightedPanelIdx((existing) => {
+          if (existing == undefined) {
+            return allFilteredPanels.length - 1;
+          }
+          const newIdx = (existing - 1) % (allFilteredPanels.length - 1);
+          return newIdx >= 0 ? newIdx : allFilteredPanels.length + newIdx;
```

Comment:
Initially I was just going to suggest a cleanup, then I noticed the `% (length - 1)` which I think is kinda wrong, but maybe doesn't matter because `existing - 1` will never be less than `-(length - 1)` unless `length <= 1`.

Actually what happens if length<=1? Can we even get here?

```suggestion
          return (existing - 1 + allFilteredPanels.length) % allFilteredPanels.length;
```